src/psyclone/psyir/nodes/whileloop.py

```diff
@@ -0,0 +1,183 @@
+        of this WhileLoop.
+
+        :returns: WhileLoop conditional expression.
+        :rtype: :py:class:`psyclone.psyir.nodes.Node`
+        :raises InternalError: If the WhileLoop node does not have the \
+            correct number of children.
+        '''
+        if len(self.children) < 2:
+            raise InternalError(
+                f"WhileLoop malformed or incomplete. It should have "
+                f"2 children, but found {len(self.children)}.")
+        return self._children[0]
+
+    @property
+    def loop_body(self):
+        ''' Return the Schedule executed when the WhileLoop condition is True.
+
+        :returns: Schedule to be executed when WhileLoop condition is True.
+        :rtype: :py:class:`psyclone.psyir.nodes.Schedule`
+        :raises InternalError: If the WhileLoop node does not have the \
+            correct number of children.
+        '''
+        if len(self.children) < 2:
+            raise InternalError(
+                f"WhileLoop malformed or incomplete. It should have "
+                f"2 children, but found {len(self.children)}.")
+
+        return self._children[1]
+
+    @staticmethod
+    def create(loop_condition, loop_body):
+        '''Create a WhileLoop instance given valid instances of an
+        loop_condition and a loop_body.
+
+        :param loop_condition: the PSyIR node containing the loop \
+            condition of the while loop statement.
+        :type loop_condition: :py:class:`psyclone.psyir.nodes.Node`
+        :param loop_body: the PSyIR nodes representing the loop body of \
+            the loop statement.
+        :type loop_body: list of :py:class:`psyclone.psyir.nodes.Node`
+
+        :returns: a WhileLoop instance.
+        :rtype: :py:class:`psyclone.psyir.nodes.WhileLoop`
+
+        :raises GenerationError: if the arguments to the create method \
+            are not of the expected type.
+
+        '''
+        if not isinstance(loop_body, list):
+            raise GenerationError(
+                "loop_body argument in create method of WhileLoop class "
+                "should be a list.")
+
+        loop_stmt = WhileLoop()
+        loop_schedule = Schedule(parent=loop_stmt)
+        loop_schedule.children = loop_body
+        loop_stmt.children = [loop_condition, loop_schedule]
+        return loop_stmt
+
+    def node_str(self, colour=True):
+        ''' Returns the name of this node with (optional) control codes
+        to generate coloured output in a terminal that supports it.
+
+        :param bool colour: whether or not to include colour control codes.
+
+        :returns: description of this node, possibly coloured.
+        :rtype: str
+        '''
+        text = self.coloured_name(colour) + "["
+        if self.annotations:
+            text += "annotations='" + ','.join(self.annotations) + "'"
+        text += "]"
+        return text
+
+    def __str__(self):
+        name = self.__class__.__name__
+        result = name + "[]\n"
```

Comment:
We don't do that for `IfBlock` though, are we sure we want to do that?

Reply:
Depending on your answer here, I'll then apply the changes to `IfBlock` in #2072.

Reply:
Oh, that's true. Personally I think it would be nicer but I the whole `__str__` thing needs revisiting anyway so I suggest you leave it as it is for this PR.